web-common/src/features/dashboards/pivot/PivotTable.svelte

```diff
@@ -124,26 +137,83 @@
       }
     }
   };
+
+  function calculateFirstColumnWidth(firstColumnName: string) {
+    const rows = $pivotDataStore.data;
+
+    const isTimeDimension = firstColumnName.startsWith("__time");
+
+    // Dates are displayed as shorter values
+    if (isTimeDimension) return MIN_COL_WIDTH;
+
+    const samples = extractSamples(
+      rows.map((row) => row[firstColumnName]),
+    ).filter((v): v is string => typeof v === "string");
+
+    const averageValueLength =
+      samples.reduce((sum, value) => sum + value.length, 0) / rows.length;
```

Comment:
Why not use the max length here and then use `Math.min(MAX_INIT_COL_WIDTH, pixelLenght + 30)`
Since we have a bound here, taking the max would make the best effort to accommodate all values until a certain width.